incubator/hnc/internal/kubectl/root.go

```diff
@@ -114,14 +116,17 @@ func homeDir() string {
 }
 
 func (cl *realClient) getHierarchy(nnm string) *api.HierarchyConfiguration {
-	if _, err := k8sClient.CoreV1().Namespaces().Get(nnm, metav1.GetOptions{}); err != nil {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
Question: what does this 5s timeout do?

Reply:
It means that if the total length of that function (eg all calls to the apiserver within `getHierarchy`) takes longer than 5s, the call will fail. Does that make sense?

Reply:
I see thanks for the explanation!